### DISCUSSION TITLE
Disable allowedPackets check in SEIP packet

### DISCUSSION THREAD

**Discussion by toberndo** [openpgpjs/openpgpjs#1617] [General]
Hey, I'm running into an issue while upgrading from v4 to v5 and that is due to the restriction of only allowing certain packets inside a [SEIP packet](https://github.com/openpgpjs/openpgpjs/blob/main/src/packet/sym_encrypted_integrity_protected_data.js#L32). We are using a SEIP packet for a private key backup and so besides a LiteralDataPacket the SEIP also includes all packets that form a private key (SecretKeyPacket,...). I didn't find a way to somehow circumvent the allowedPackets check and wonder if the only way would be to add another config parameter.

Currently there is:
```
ignoreUnsupportedPackets: true,
ignoreMalformedPackets: false,
```

Would it be reasonable to have something like `disableAllowedPacketsCheck: false`?

**Comment by twiss** [comment]
Hey :wave: It's actually not super simple to add such a config option because the SEIP packet imports (only) the packets it can contain, and the packet parser by itself doesn't actually know about (other) packets anymore. This was done to allow tree shaking, e.g. if you only import functions related to signatures, then you won't need the AES code, etc. 

So really what we'd need is a config option to pass additional allowed packet classes (which you'd then first need to import yourself), e.g. `additionalAllowedPackets`. Then, if you really wanted to allow all of them (and you don't care about tree shaking, e.g. you're importing the entirety of openpgp.js anyway) you could pass something like `additionalAllowedPackets: util.constructAllowedPackets(Object.values(openpgp).filter(packetClass => !!packetClass.tag))`, but probably it'd be better to pass the actual list of packets you need.

**Comment by toberndo** [comment]
@twiss Thanks for pointing in this direction of the additionalAllowedPackets config option! I created pull request https://github.com/openpgpjs/openpgpjs/pull/1618 that implements the concept.

**Reply by twiss** [reply]
Thanks!
